sdk/compiler/damlc/lib/DA/Cli/Options.hs

```diff
@@ -591,17 +593,21 @@ optionsParser numProcessors enableScenarioService parsePkgName parseDlintUsage =
 
     optWarnBadInterfaceInstances :: Parser Bool
     optWarnBadInterfaceInstances =
-      flagYesNoAuto
-        "warn-bad-interface-instances"
-        defaultUiWarnBadInterfaceInstances
-        "Convert errors about bad, non-upgradeable interface instances into warnings."
-        idm
+      determineAuto defaultUiWarnBadInterfaceInstances <$>
+        flagYesNoAuto''
+          "-Wupgrade-interfaces"
+          "Convert errors about bad, non-upgradeable interface instances into warnings."
+          idm
```

Comment:
Why not have this flag build a RawWarningFlag, rather than keeping that information all the way down the stack?

Reply:
Per discussion on call - we'd like to preserve this info to enable future deprecation warnings